utils/metrics.py

```diff
@@ -0,0 +1,192 @@
+import itertools
+
+import numpy as np
+import ot
+import polars as pl
+from scipy.spatial.distance import cdist
+
+
+def _generate_on_off_profiles(
+    profiles: pl.DataFrame, on_signature: list[str], off_signature: list[str]
+):
+    """Generate on and off profiles from the given profiles.
+
+    Parameters
+    ----------
+    profiles : pl.DataFrame
+        The input profiles DataFrame.
+    on_signature : list[str]
+        Morphological profiles that are in the on-morphology signature.
+    off_signature : list[str]
+        The list of features to include in the off profile.
+
+    Returns
+    -------
+    tuple
+        A tuple containing two DataFrames: the on profile and the off profile.
+    """
+    on_profiles = profiles[on_signature]
+    off_profiles = profiles[off_signature]
+    return on_profiles, off_profiles
+
+
+def earths_movers_distance(
+    ref_profiles: pl.DataFrame,
+    exp_profiles: pl.DataFrame,
+    on_signature: list[str],
+    off_signature: list[str],
+    distance_metric: str = "euclidean",
+):
+    """Compute the Earth Mover's Distance (EMD) between reference and experimental profiles.
+
+    Takes in the reference and experimental profiles, along with their on and off
+    signatures, and computes the EMD. Two scores will be returned: the EMD for the
+    on-morphology profiles and the EMD for the off-morphology profiles.
+
+    Parameters
+    ----------
+    ref_profiles : pl.DataFrame
+        The reference profiles DataFrame.
+    exp_profiles : pl.DataFrame
+        The experimental profiles DataFrame.
+    on_signature : list[str]
+        Morphological profiles that are in the on-morphology signature.
+    off_signature : list[str]
+        The list of features to include in the off profile.
+    distance_metric : str, optional
+        Distance metric to use when generating the distance matrices.
+    Returns
```

Comment:
Consider adding an additional space here. to keep consistent with the docs style.
```suggestion
        Distance metric to use when generating the distance matrices.
        
    Returns
```